### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/rust-clippy/actions/runs/4710771873#summary-12776860721

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/rust-clippy/actions/runs/4710771874#summary-12776860722

changelog: none

### CODE REVIEW

File: clippy_lints/src/manual_assert.rs [hunk 1]
```diff
@@ -64,7 +64,7 @@ impl<'tcx> LateLintPass<'tcx> for ManualAssert {
             };
             let cond_sugg = sugg::Sugg::hir_with_applicability(cx, cond, "..", &mut applicability).maybe_par();
             let sugg = format!("assert!({not}{cond_sugg}, {format_args_snip});");
-            // we show to the user the suggestion without the comments, but when applicating the fix, include the comments in the block
+            // we show to the user the suggestion without the comments, but when applying the fix, include the comments in the block
```
Comment (jsoref) [inline clippy_lints/src/manual_assert.rs:67]: This is a bit of an atypical change. happy to drop.

File: clippy_lints/src/trailing_empty_array.rs [hunk 2]
```diff
@@ -60,7 +60,7 @@ fn is_struct_with_trailing_zero_sized_array(cx: &LateContext<'_>, item: &Item<'_
         if let Some(last_field) = data.fields().last();
         if let rustc_hir::TyKind::Array(_, rustc_hir::ArrayLen::Body(length)) = last_field.ty.kind;
 
-        // Then check if that that array zero-sized
+        // Then check if that array is zero-sized
```
Comment (jsoref) [inline clippy_lints/src/trailing_empty_array.rs:63]: this is an atypical correction

File: clippy_lints/src/types/mod.rs [hunk 3]
```diff
@@ -90,7 +90,7 @@ declare_clippy_lint! {
     ///
     /// ### Why is this bad?
     /// `Option<_>` represents an optional value. `Option<Option<_>>`
-    /// represents an optional optional value which is logically the same thing as an optional
+    /// represents an optional value which is logically the same thing as an optional
```
Comment (jsoref) [inline clippy_lints/src/types/mod.rs:93]: This change is technically, wrong, but there should be a way to clarify, how about:
```suggestion
    /// represents an optional value which itself is optional, and this is logically the same thing as an optional
```

File: clippy_lints/src/wildcard_imports.rs [hunk 4]
```diff
@@ -143,7 +143,7 @@ impl LateLintPass<'_> for WildcardImports {
                 } else {
                     // In this case, the `use_path.span` ends right before the `::*`, so we need to
                     // extend it up to the `*`. Since it is hard to find the `*` in weird
-                    // formattings like `use _ ::  *;`, we extend it up to, but not including the
+                    // forms like `use _ ::  *;`, we extend it up to, but not including the
```
Comment (jsoref) [inline clippy_lints/src/wildcard_imports.rs:146]: happy to drop if people are really attached to it.

File: clippy_utils/src/lib.rs [hunk 5]
```diff
@@ -2234,7 +2234,7 @@ pub fn is_slice_of_primitives(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<S
 }
 
 /// returns list of all pairs (a, b) from `exprs` such that `eq(a, b)`
-/// `hash` must be comformed with `eq`
+/// `hash` must be verified with `eq`
```
Comment (jsoref) [inline clippy_utils/src/lib.rs:2237]: The original word was probably `confirmed`, but `verified` is better (someone else suggested it to me and I agree)

File: clippy_utils/src/ty.rs [hunk 6]
```diff
@@ -1101,7 +1101,7 @@ pub fn make_projection<'tcx>(
 ///
 /// This function is for associated types which are "known" to be valid with the given
 /// substitutions, and as such, will only return `None` when debug assertions are disabled in order
-/// to prevent ICE's. With debug assertions enabled this will check that that type normalization
+/// to prevent ICE's. With debug assertions enabled this will check that type normalization
```
Comment (jsoref) [inline clippy_utils/src/ty.rs:1104]: sometimes `that that` is intentional, and occasionally it's necessary, I don't think it's necessary here, and in fact, I suspect that in this case the second `that` wouldn't properly reference a thing that is type normalization.

File: tests/ui/crashes/ice_exact_size.rs
Comment (jsoref) [inline tests/ui/crashes/ice_exact_size.rs:1]:
```diff
-exacte
+exact
```

File: tests/ui/no_mangle_with_rust_abi.rs [hunk 7]
```diff
@@ -25,7 +25,7 @@ fn rust_abi_multiline_function_really_long_name_to_overflow_args_to_multiple_lin
     0
 }
 
-// Must not run on functions that explicitly opt in to Rust ABI with `extern "Rust"`
+// Must not run on functions that explicitly opt into Rust ABI with `extern "Rust"`
```
Comment (jsoref) [inline tests/ui/no_mangle_with_rust_abi.rs:28]: `in to`/`into` has been quite contentious, happy to drop...

File: tests/ui/option_if_let_else.fixed [hunk 8]
```diff
@@ -97,7 +97,7 @@ enum DummyEnum {
     Two,
 }
 
-// should not warn since there is a compled complex subpat
+// should not warn since there is a complex subpat
```
Comment (jsoref) [inline tests/ui/option_if_let_else.fixed:100]: i think that `compled` was a typo for `complex` and then `complex` was added again -- that only a single `complex` was intended. But I could be wrong.

File: tests/ui/same_name_method.rs [hunk 9]
```diff
@@ -62,7 +62,7 @@ mod should_lint {
         impl T1 for S {}
     }
 
-    mod multiply_conflicit_trait {
+    mod multiply_conflict_trait {
```
Comment (jsoref) [inline tests/ui/same_name_method.rs:65]: ?

File: util/gh-pages/index.html [hunk 10]
```diff
@@ -564,7 +564,7 @@ <h2 class="panel-title">
     </div>
 
     <a href="https://github.com/rust-lang/rust-clippy">
-        <img style="position: absolute; top: 0; right: 0; border: 0; clip-path: polygon(0% 0%, 100% 0%, 100% 100%);" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on Github"/>
+        <img style="position: absolute; top: 0; right: 0; border: 0; clip-path: polygon(0% 0%, 100% 0%, 100% 100%);" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"/>
```
Comment (jsoref) [inline util/gh-pages/index.html:567]: Brand

**Review by jsoref — COMMENTED** [review]
Most spelling corrections automatically suggested by Google Sheets (when being run on the rust repository). I've manually transplanted these commits into this repository.

All fault mine.

File: clippy_lints/src/loops/manual_memcpy.rs [hunk 11]
```diff
@@ -15,7 +15,7 @@ use rustc_span::symbol::sym;
 use std::fmt::Display;
 use std::iter::Iterator;
 
-/// Checks for for loops that sequentially copy items from one slice-like
+/// Checks for loops that sequentially copy items from one slice-like
```
Comment (Jules-Bertholet) [inline clippy_lints/src/loops/manual_memcpy.rs:18]: False positive? ("Checks for `for` loops")

File: clippy_lints/src/formatting.rs [hunk 12]
```diff
@@ -10,7 +10,7 @@ use rustc_span::source_map::Span;
 
 declare_clippy_lint! {
     /// ### What it does
-    /// Checks for usage of the non-existent `=*`, `=!` and `=-`
+    /// Checks for usage of the nonexistent `=*`, `=!` and `=-`
```
Comment (Jarcho) [inline clippy_lints/src/formatting.rs:13]: Both the hyphenated and non-hyphenated versions are used. No need to remove the change, but the spell checker is overzealous.

File: clippy_lints/src/manual_assert.rs (hunk 1 above)
Comment (Jarcho) [reply clippy_lints/src/manual_assert.rs:67]: 'applicating' is fine, but unusual. The change can stay.

File: clippy_utils/src/ty.rs (hunk 6 above)
Comment (Jarcho) [reply clippy_utils/src/ty.rs:1104]: As the person who wrote that, that is a typo. One I do way too frequently.

File: tests/ui/needless_for_each_fixable.fixed [hunk 13]
```diff
@@ -110,7 +110,7 @@ fn should_not_lint() {
         }),
     }
 
-    // `for_each` is in a let bingind.
+    // `for_each` is in a let bingint.
```
Comment (Jarcho) [inline tests/ui/needless_for_each_fixable.fixed:113]: Should be 'binding'.

File: tests/ui/needless_for_each_fixable.rs [hunk 14]
```diff
@@ -110,7 +110,7 @@ fn should_not_lint() {
         }),
     }
 
-    // `for_each` is in a let bingind.
+    // `for_each` is in a let bingint.
```
Comment (Jarcho) [inline tests/ui/needless_for_each_fixable.rs:113]: Should be 'binding'.

File: tests/ui/no_mangle_with_rust_abi.rs (hunk 7 above)
Comment (Jarcho) [reply tests/ui/no_mangle_with_rust_abi.rs:28]: Should be 'opt in to using the Rust ABI' if you want to be more formal.

File: tests/ui/same_name_method.rs (hunk 9 above)
Comment (Jarcho) [reply tests/ui/same_name_method.rs:65]: Should be 'multiple_conflicting_traits'.

File: clippy_lints/src/wildcard_imports.rs (hunk 4 above)
Comment (Jarcho) [reply clippy_lints/src/wildcard_imports.rs:146]: 'formattings' makes it clear that it's about how the thing is formatted. The awkward spacing in example being the only weird thing _should_ make that apparent, but it's possible to not pick up on that right away.

Given that the only benefit here is being six characters shorter, it's better to revert this.

File: clippy_lints/src/formatting.rs (hunk 12 above)
Comment (jsoref) [reply clippy_lints/src/formatting.rs:13]: That's actually me, the human. I have some preferences. My main preference is for consistency (it's much easier to search and get all matches than search and get <1/2 of matches - as there''s often a distribution between two accepted spellings and some misspellings of the same), and my secondary preference is to not hyphenate.

I'm certainly willing to drop if people have opinions.

File: tests/ui/needless_for_each_fixable.fixed (hunk 13 above)
Comment (jsoref) [reply tests/ui/needless_for_each_fixable.fixed:113]: Sigh. I somehow read it as a typo of bigint. I totally missed that the `n` there hinted at it being a letter swap.

File: clippy_utils/src/lib.rs [hunk 15]
```diff
@@ -2234,7 +2234,7 @@ pub fn is_slice_of_primitives(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<S
 }
 
 /// returns list of all pairs (a, b) from `exprs` such that `eq(a, b)`
-/// `hash` must be comformed with `eq`
+/// implies `hash(a) == hash(b)`
```
Comment (Jarcho) [inline clippy_utils/src/lib.rs:2237]: Should be 'Requires `eq` and `hash` be functions such that `eq(a, b)` implies `hash(a) == hash(b)`.' It's a precondition to the function, not a description of what the function does.

File: tests/ui/no_mangle_with_rust_abi.rs [hunk 16]
```diff
@@ -25,7 +25,7 @@ fn rust_abi_multiline_function_really_long_name_to_overflow_args_to_multiple_lin
     0
 }
 
-// Must not run on functions that explicitly opt in to Rust ABI with `extern "Rust"`
+// Must not run on functions that explicitly opt in to using Rust ABI with `extern "Rust"`
```
Comment (Jarcho) [inline tests/ui/no_mangle_with_rust_abi.rs:28]: 'using the'

File: clippy_utils/src/lib.rs (hunk 15 above)
Comment (jsoref) [reply clippy_utils/src/lib.rs:2237]: I took a stab at it.

Comment (Jarcho) [reply clippy_utils/src/lib.rs:2237]: Just flip the two paragraphs and then it's fine. Weird to have details of how to use the function before knowing what it does.

Sorry for the delay here. Been busy this week.

**Comment by jsoref** [comment]
Swapped

File: clippy_utils/src/lib.rs (hunk 15 above)
Comment (jsoref) [reply clippy_utils/src/lib.rs]: I'm in no hurry and I appreciate that people are taking time from their normal work to review these changes.

I've squashed all the spelling changes into a single commit and left this change as its own, but if you'd like, I can squash it in as well.

**Comment by Jarcho** [comment]
Thank you. @bors r+

File: clippy_utils/src/lib.rs (hunk 15 above)
Comment (Jarcho) [reply clippy_utils/src/lib.rs]: It works as a separate commit.

**Pull request merged**